### ISSUE TITLE
Random API Errors

### ISSUE THREAD

**Issue by Marcos-kp** [fondberg/spotcast#367]
When trying to play on my Echo device, I get this error: http status: 404, code:-1 - https://api.spotify.com/v1/me/player/play?device_id=fe4b77b1-a5b0-4814-8188-d3acf61ds5c6_amzn_1: Not found., reason: None
Sometimes the service works, sometimes it doesn't. I noticed that if I call the service twice in a roll, the chances of working are higher, but not 100%

I also tried to use or the entity id or device name, but both gave me "Could not find device with name" error.
(I changed my device id for privacy)

**Comment by TheMrDec** [comment]
I am also seeing inconsistent 404s for the PUT request used to start the player. The strange thing is that if I use curl to manually send the GET and PUT that spotipy is using, it works 100% of the time.
@fondberg is it possible to have the HTTP requests output to a logfile so that we can verify that they are not malformed? I am not terribly familiar with the HA logging setup or I would just fork and add that to see what happens. Might still do that just to serve my curiosity.

**Comment by mrkoopa1** [comment]
This may be what I am hitting aswell, I thought this latest release would solve it as in the group I have one pretty slow Lenovo clock that has google built in. But alas it is still here, its hit or miss, sometimes it works sometimes it kicks the below debug logs out. Could this potentially be the same thing?
```
2023-02-04 10:37:09.131 INFO (MainThread) [homeassistant.components.script.play_chilledcow] Play_chilledcow: Running script sequence
2023-02-04 10:37:09.131 INFO (MainThread) [homeassistant.components.script.play_chilledcow] Play_chilledcow: Executing step call service
2023-02-04 10:37:09.132 DEBUG (SyncWorker_1) [custom_components.spotcast.spotcast_controller] setting up with  account default
2023-02-04 10:37:09.132 DEBUG (SyncWorker_1) [custom_components.spotcast.spotcast_controller] expires: 1675531873 time: 1675528629.132378
2023-02-04 10:37:09.132 DEBUG (SyncWorker_1) [custom_components.spotcast.spotcast_controller] setting up with  account default
023-02-04 10:37:09.289 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_spotify_devices: media_player.spotify_mrkoopa: None: []
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_spotify_devices: {'devices': []}
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.living_room_tv: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-ccc12a06bb0e244a7eefa7a67c50657a._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.176', 8009))}, uuid=UUID('ccc12a06-bb0e-244a-7eef-a7a67c50657a'), model_name='Chromecast', friendly_name='Guest Bedroom', host='172.16.10.176', port=8009, cast_type='cast', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.googlehome6826: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-Mini-af3baba68c30fccbc16146f3c7d6ba15._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.171', 8009))}, uuid=UUID('af3baba6-8c30-fccb-c161-46f3c7d6ba15'), model_name='Google Home Mini', friendly_name='Downstairs bath speaker', host='172.16.10.171', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.mrkoopa_bedroom_clock: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.159', 8009)), ServiceInfo(type='mdns', data='Lenovo-Smart-Clock-afd02c1dc337774472a0e47292480b72._googlecast._tcp.local.')}, uuid=UUID('afd02c1d-c337-7744-72a0-e47292480b72'), model_name='Lenovo Smart Clock', friendly_name='Master Bed Clock', host='172.16.10.159', port=8009, cast_type='audio', manufacturer='LENOVO'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.chill: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Cast-Group-73931280aaa64c1bb714982942e24086._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.159', 32041))}, uuid=UUID('73931280-aaa6-4c1b-b714-982942e24086'), model_name='Google Cast Group', friendly_name='Chill', host='172.16.10.159', port=32041, cast_type='group', manufacturer='Google Inc.'), is_dynamic_group=False)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.googlehome3901: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.101', 8009)), ServiceInfo(type='mdns', data='Google-Nest-Mini-d7be16f755c2d981f4f9dd542ff789ef._googlecast._tcp.local.')}, uuid=UUID('d7be16f7-55c2-d981-f4f9-dd542ff789ef'), model_name='Google Nest Mini', friendly_name='Guest Bathroom', host='172.16.10.101', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.chromecastultra8670: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-Ultra-700dddce67c915a3029a7ff7cd9aa596._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.165', 8009))}, uuid=UUID('700dddce-67c9-15a3-029a-7ff7cd9aa596'), model_name='Chromecast Ultra', friendly_name='Living room TV', host='172.16.10.165', port=8009, cast_type='cast', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.nestmini8314: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Nest-Mini-4ec2fb7d168f1dbd2d564e829cd6159f._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.173', 8009))}, uuid=UUID('4ec2fb7d-168f-1dbd-2d56-4e829cd6159f'), model_name='Google Nest Mini', friendly_name='War Room speaker', host='172.16.10.173', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.nestaudio8413: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.122', 8009)), ServiceInfo(type='mdns', data='Nest-Audio-f3ed6d6abdb24a3c104abdee3c26ddca._googlecast._tcp.local.')}, uuid=UUID('f3ed6d6a-bdb2-4a3c-104a-bdee3c26ddca'), model_name='Nest Audio', friendly_name='Living Room speaker', host='172.16.10.122', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.nestaudio0239: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.121', 8009)), ServiceInfo(type='mdns', data='Nest-Audio-bd7b3e7bd337d2188a4e1d17da513c43._googlecast._tcp.local.')}, uuid=UUID('bd7b3e7b-d337-d218-8a4e-1d17da513c43'), model_name='Nest Audio', friendly_name='Kitchen speaker', host='172.16.10.121', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.343 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.hi_fi: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.121', 32170)), ServiceInfo(type='mdns', data='Google-Cast-Group-d6b742079f224fb59d6203d13e72ad95-1._googlecast._tcp.local.')}, uuid=UUID('d6b74207-9f22-4fb5-9d62-03d13e72ad95'), model_name='Google Cast Group', friendly_name='Hi-Fi', host='172.16.10.121', port=32170, cast_type='group', manufacturer='Google Inc.'), is_dynamic_group=False)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.googlehome0122: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.174', 8009)), ServiceInfo(type='mdns', data='Google-Home-Mini-0b354d9b6e319c169251cf130c70124f._googlecast._tcp.local.')}, uuid=UUID('0b354d9b-6e31-9c16-9251-cf130c70124f'), model_name='Google Home Mini', friendly_name='Garage speaker', host='172.16.10.174', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.war_room_mini: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.167', 8009)), ServiceInfo(type='mdns', data='Google-Home-Mini-25bf856086ece35b224691c56e895526._googlecast._tcp.local.')}, uuid=UUID('25bf8560-86ec-e35b-2246-91c56e895526'), model_name='Google Home Mini', friendly_name='Guest Bed Mini', host='172.16.10.167', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.living_room_chromecast: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-113c9a692d9f331020f3106412cb966d._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.100', 8009))}, uuid=UUID('113c9a69-2d9f-3310-20f3-106412cb966d'), model_name='Chromecast', friendly_name='Master Bed Chromecast', host='172.16.10.100', port=8009, cast_type='cast', manufacturer='Google'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.party_group: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Cast-Group-2b2df5fc8bce45aa99824b9a16d54cd1-1._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.100', 32083))}, uuid=UUID('2b2df5fc-8bce-45aa-9982-4b9a16d54cd1'), model_name='Google Cast Group', friendly_name='Party Group', host='172.16.10.100', port=32083, cast_type='group', manufacturer='Google Inc.'), is_dynamic_group=False)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.googlehome2913: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-a7e8c2ea6fd8d14b0287ff118d26d400._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.189', 8009))}, uuid=UUID('a7e8c2ea-6fd8-d14b-0287-ff118d26d400'), model_name='Google Home', friendly_name='Dining Room speaker', host='172.16.10.189', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.kitchen_display: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.188', 8009)), ServiceInfo(type='mdns', data='google-nest-hub-23d11a140807f4190d359780b9027e6e._googlecast._tcp.local.')}, uuid=UUID('23d11a14-0807-f419-0d35-9780b9027e6e'), model_name='Google Nest Hub', friendly_name='Kitchen display', host='172.16.10.188', port=8009, cast_type='cast', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.googlehome1699: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-19d04e0c03a1b960a4080ea4ab75c8ed._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.170', 8009))}, uuid=UUID('19d04e0c-03a1-b960-a408-0ea4ab75c8ed'), model_name='Google Home', friendly_name='Master Bathroom speaker', host='172.16.10.170', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.googlehome9780: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-Mini-517e12d3ade67830a77445b4d86deaa3._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.166', 8009))}, uuid=UUID('517e12d3-ade6-7830-a774-45b4d86deaa3'), model_name='Google Home Mini', friendly_name='Media Room speaker', host='172.16.10.166', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.helpers] get_cast_devices: media_player.chromecast: None cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-03536db6b424261800f886249a526487._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.114', 8009))}, uuid=UUID('03536db6-b424-2618-00f8-86249a526487'), model_name='Chromecast', friendly_name='Media Room TV', host='172.16.10.114', port=8009, cast_type='cast', manufacturer='Google'), is_dynamic_group=None)
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.spotcast_controller] Chromecast devices: [ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-ccc12a06bb0e244a7eefa7a67c50657a._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.176', 8009))}, uuid=UUID('ccc12a06-bb0e-244a-7eef-a7a67c50657a'), model_name='Chromecast', friendly_name='Guest Bedroom', host='172.16.10.176', port=8009, cast_type='cast', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-Mini-af3baba68c30fccbc16146f3c7d6ba15._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.171', 8009))}, uuid=UUID('af3baba6-8c30-fccb-c161-46f3c7d6ba15'), model_name='Google Home Mini', friendly_name='Downstairs bath speaker', host='172.16.10.171', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.159', 8009)), ServiceInfo(type='mdns', data='Lenovo-Smart-Clock-afd02c1dc337774472a0e47292480b72._googlecast._tcp.local.')}, uuid=UUID('afd02c1d-c337-7744-72a0-e47292480b72'), model_name='Lenovo Smart Clock', friendly_name='Master Bed Clock', host='172.16.10.159', port=8009, cast_type='audio', manufacturer='LENOVO'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Cast-Group-73931280aaa64c1bb714982942e24086._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.159', 32041))}, uuid=UUID('73931280-aaa6-4c1b-b714-982942e24086'), model_name='Google Cast Group', friendly_name='Chill', host='172.16.10.159', port=32041, cast_type='group', manufacturer='Google Inc.'), is_dynamic_group=False), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.101', 8009)), ServiceInfo(type='mdns', data='Google-Nest-Mini-d7be16f755c2d981f4f9dd542ff789ef._googlecast._tcp.local.')}, uuid=UUID('d7be16f7-55c2-d981-f4f9-dd542ff789ef'), model_name='Google Nest Mini', friendly_name='Guest Bathroom', host='172.16.10.101', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-Ultra-700dddce67c915a3029a7ff7cd9aa596._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.165', 8009))}, uuid=UUID('700dddce-67c9-15a3-029a-7ff7cd9aa596'), model_name='Chromecast Ultra', friendly_name='Living room TV', host='172.16.10.165', port=8009, cast_type='cast', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Nest-Mini-4ec2fb7d168f1dbd2d564e829cd6159f._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.173', 8009))}, uuid=UUID('4ec2fb7d-168f-1dbd-2d56-4e829cd6159f'), model_name='Google Nest Mini', friendly_name='War Room speaker', host='172.16.10.173', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.122', 8009)), ServiceInfo(type='mdns', data='Nest-Audio-f3ed6d6abdb24a3c104abdee3c26ddca._googlecast._tcp.local.')}, uuid=UUID('f3ed6d6a-bdb2-4a3c-104a-bdee3c26ddca'), model_name='Nest Audio', friendly_name='Living Room speaker', host='172.16.10.122', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.121', 8009)), ServiceInfo(type='mdns', data='Nest-Audio-bd7b3e7bd337d2188a4e1d17da513c43._googlecast._tcp.local.')}, uuid=UUID('bd7b3e7b-d337-d218-8a4e-1d17da513c43'), model_name='Nest Audio', friendly_name='Kitchen speaker', host='172.16.10.121', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.121', 32170)), ServiceInfo(type='mdns', data='Google-Cast-Group-d6b742079f224fb59d6203d13e72ad95-1._googlecast._tcp.local.')}, uuid=UUID('d6b74207-9f22-4fb5-9d62-03d13e72ad95'), model_name='Google Cast Group', friendly_name='Hi-Fi', host='172.16.10.121', port=32170, cast_type='group', manufacturer='Google Inc.'), is_dynamic_group=False), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.174', 8009)), ServiceInfo(type='mdns', data='Google-Home-Mini-0b354d9b6e319c169251cf130c70124f._googlecast._tcp.local.')}, uuid=UUID('0b354d9b-6e31-9c16-9251-cf130c70124f'), model_name='Google Home Mini', friendly_name='Garage speaker', host='172.16.10.174', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.167', 8009)), ServiceInfo(type='mdns', data='Google-Home-Mini-25bf856086ece35b224691c56e895526._googlecast._tcp.local.')}, uuid=UUID('25bf8560-86ec-e35b-2246-91c56e895526'), model_name='Google Home Mini', friendly_name='Guest Bed Mini', host='172.16.10.167', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-113c9a692d9f331020f3106412cb966d._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.100', 8009))}, uuid=UUID('113c9a69-2d9f-3310-20f3-106412cb966d'), model_name='Chromecast', friendly_name='Master Bed Chromecast', host='172.16.10.100', port=8009, cast_type='cast', manufacturer='Google'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Cast-Group-2b2df5fc8bce45aa99824b9a16d54cd1-1._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.100', 32083))}, uuid=UUID('2b2df5fc-8bce-45aa-9982-4b9a16d54cd1'), model_name='Google Cast Group', friendly_name='Party Group', host='172.16.10.100', port=32083, cast_type='group', manufacturer='Google Inc.'), is_dynamic_group=False), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-a7e8c2ea6fd8d14b0287ff118d26d400._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.189', 8009))}, uuid=UUID('a7e8c2ea-6fd8-d14b-0287-ff118d26d400'), model_name='Google Home', friendly_name='Dining Room speaker', host='172.16.10.189', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='host', data=('172.16.10.188', 8009)), ServiceInfo(type='mdns', data='google-nest-hub-23d11a140807f4190d359780b9027e6e._googlecast._tcp.local.')}, uuid=UUID('23d11a14-0807-f419-0d35-9780b9027e6e'), model_name='Google Nest Hub', friendly_name='Kitchen display', host='172.16.10.188', port=8009, cast_type='cast', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-19d04e0c03a1b960a4080ea4ab75c8ed._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.170', 8009))}, uuid=UUID('19d04e0c-03a1-b960-a408-0ea4ab75c8ed'), model_name='Google Home', friendly_name='Master Bathroom speaker', host='172.16.10.170', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Home-Mini-517e12d3ade67830a77445b4d86deaa3._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.166', 8009))}, uuid=UUID('517e12d3-ade6-7830-a774-45b4d86deaa3'), model_name='Google Home Mini', friendly_name='Media Room speaker', host='172.16.10.166', port=8009, cast_type='audio', manufacturer='Google Inc.'), is_dynamic_group=None), ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Chromecast-03536db6b424261800f886249a526487._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.114', 8009))}, uuid=UUID('03536db6-b424-2618-00f8-86249a526487'), model_name='Chromecast', friendly_name='Media Room TV', host='172.16.10.114', port=8009, cast_type='cast', manufacturer='Google'), is_dynamic_group=None)]
2023-02-04 10:37:09.344 DEBUG (SyncWorker_1) [custom_components.spotcast.spotcast_controller] cast info: ChromecastInfo(cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Cast-Group-73931280aaa64c1bb714982942e24086._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.159', 32041))}, uuid=UUID('73931280-aaa6-4c1b-b714-982942e24086'), model_name='Google Cast Group', friendly_name='Chill', host='172.16.10.159', port=32041, cast_type='group', manufacturer='Google Inc.'), is_dynamic_group=False)
2023-02-04 10:37:09.345 DEBUG (SyncWorker_1) [custom_components.spotcast.spotcast_controller] Found cast device: Chromecast('unknown', port=8009, cast_info=CastInfo(services={ServiceInfo(type='mdns', data='Google-Cast-Group-73931280aaa64c1bb714982942e24086._googlecast._tcp.local.'), ServiceInfo(type='host', data=('172.16.10.159', 32041))}, uuid=UUID('73931280-aaa6-4c1b-b714-982942e24086'), model_name='Google Cast Group', friendly_name='Chill', host='172.16.10.159', port=32041, cast_type='group', manufacturer='Google Inc.'))
2023-02-04 10:37:09.500 INFO (SyncWorker_1) [pychromecast.controllers] Receiver:Launching app CC32E753
2023-02-04 10:37:19.133 INFO (MainThread) [homeassistant.components.script.play_chilledcow] Play_chilledcow: Executing step call service
2023-02-04 10:37:19.135 INFO (SyncWorker_6) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.136 INFO (SyncWorker_8) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.136 INFO (SyncWorker_3) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.136 INFO (SyncWorker_8) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.138 INFO (SyncWorker_2) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.138 INFO (SyncWorker_10) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.138 INFO (SyncWorker_2) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.139 INFO (SyncWorker_10) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.139 INFO (SyncWorker_6) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:19.139 INFO (SyncWorker_4) [pychromecast.controllers] Receiver:setting volume to 0.4
2023-02-04 10:37:20.502 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall spotcast.start (c:01GREMKTW9G9MX6FF5BGNQ8MV1): uri=spotify:playlist:0vvXsWCC9xrXsKd4FyS8kM?si=7830c1e742d5441b, random_song=True, shuffle=True, start_volume=40, entity_id=media_player.chill, ignore_fully_played=False, force_playback=False, offset=0, limit=20, repeat=off>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1805, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1830, in _execute_service
    await self._hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/spotcast/__init__.py", line 221, in start_casting
    spotify_device_id = spotcast_controller.get_spotify_device_id(
  File "/config/custom_components/spotcast/spotcast_controller.py", line 250, in get_spotify_device_id
    spotify_cast_device.startSpotifyController(access_token, expires)
  File "/config/custom_components/spotcast/spotcast_controller.py", line 95, in startSpotifyController
    sp.launch_app()
  File "/config/custom_components/spotcast/spotify_controller.py", line 104, in launch_app
    raise LaunchError(
pychromecast.error.LaunchError: Timeout when waiting for status response from Spotify app
```

**Comment by fcusson** [comment]
this issue should be resolved in `v3.7.0`. Please repoen if still present
